### PR TITLE
codecov configs

### LINKED ISSUE #4440: [Maintenance] Codecov results have been inconsistent
### Description and related issues
We are seeing large reductions and increases in Codecov results in GitHub. When you check the results on Codecov they look correct, but there is a warning about "Commits have different number of coverage report uploads" that seems to be causing the problem. Investigate how to make the numbers reflect actual changes in code coverage caused by the PR

https://app.codecov.io/gh/Enterprise-CMCS/eAPD

The Codecov warning links to these resources:
* https://docs.codecov.com/docs/unexpected-coverage-changes#mismatching-base-and-head-commit-upload-counts
* https://youtu.be/QyXk6QwAXk8

### This task is done when…

- [ ] the Codecov section of the PR reflects the actual changes

### PR DESCRIPTION
**Pull request by thetif**
Resolves #4440 

### Description
Cleaning up the codecov configs. Found two api unit tests that were causing errors and renamed them to *.brokentest.js. I added them to #4516 to look into further. I was able to fix the codecov issue with the api unit tests not reporting the correct coverage. I changed the endpoint tests to set up and tear down mongo with each test suite, like we are doing for the postgres db; and I updated the way we are setting up the axios calls to only set up one axios const for each user type and reusing those in every test. I think we were creating too many axios consts and they weren't being closed once the tests were done.

### This pull request is ready to code review when

- [ ] Pull request has been labeled, if applicable with feature, content, bug,
      tests, refactor

### This pull request is ready to test when

- [ ] Code has been reviewed by someone other than the original author

### This pull request is ready to review when the QA has

- [ ] Verified the functionality related to the change
- [ ] Verified that the change works with Narrator on Windows
- [ ] Verified that the change works with VoiceOver on Mac
- [ ] Verified all updated pages with the WAVE tool
- [ ] Verified tab and keyboard navigation functionality

### This pull request can be merged when

- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: web/jest.config.js [hunk 1]
```diff
@@ -15,15 +14,16 @@ module.exports = {
   testEnvironment: 'jsdom',
   snapshotSerializers: ['enzyme-to-json/serializer'],
   transform: {
-    '^.+\\.jsx?$': 'babel-jest',
+    '^.+\\.(t|j)sx?$': 'babel-jest',
```
Comment (tbolt) [inline web/jest.config.js:17]: 👀

**Review by tbolt — APPROVED** [review]

File: web/jest.config.js (hunk 1 above)
Comment (amyd11) [reply web/jest.config.js:17]: Typescript foreshadowing?!

**Review by amyd11 — APPROVED** [review]

**Review by tbolt — APPROVED** [review]

**Review by Sun-Mountain — APPROVED** [review]
🚀

**Comment by thetif** [comment]
Merging without QA and Project Review because the PR is entirely automated test and CI/CD config updates.

**Pull request merged**